Week-1/tanmay-sinha/app.py

```diff
@@ -0,0 +1,215 @@
+       
+        if len(url) == 0:
+            URL = input(f"Enter the URL from 91mobiles.com only (for better results):\n")
+        else:
+            URL = url
+        try:
+            page = requests.get(URL)
+            if page.status_code ==200:
+                print("Correct URL")
+                self.URL = URL
+                self.scraping_site(page)
+            else:
+                print('Wrong URL!! Enter correct URL:')
+        except:
+            print('Wrong URL!! Enter correct URL:')  
+         
+        
+    def scraping_site(self, page):
+        soup = BeautifulSoup(page.content, 'html.parser')
+        features = soup.find_all("td", {"class":"spec_ttle"})
+        values = soup.find_all("td", {"class":"spec_des"})
+        phone_name = soup.find("span", {"itemprop":"name"}).get_text()
+        stores = soup.find_all("img", {"class":"img_alt"})
+        prices = soup.find_all("span", {"class":"prc"})
+        # scraping the details and making dictionary.
+        self.phone_name = phone_name
+        dictionary ={}
+        dictionary['URL'] = self.URL
+        for i in range(len(features)):
+            feature = features[i].text.replace("\n", "").strip()
+            value = values[i].text.replace("\n", "").strip()
+
+            if '  ' in value:
+                value = value[0:value.find('  ')]
+
+            dictionary[feature] = value
+
+            if len(value) is 0:
+                if 'yes' in str(values[i]):
+                    dictionary[feature] = 'Yes'
+                else:
+                    dictionary[feature] = 'No'
+
+        #scraping the price and putting it in dictionary.
+        dictionary['prices'] ={}
+        for i in range(len(prices)):
+            store = stores[i]['alt']
+            price = prices[i].get_text().replace("\n","")
+            dictionary['prices'][store] = price
+
+        jsonstr = json.dumps(dictionary,sort_keys=True,indent=4)
+        self.make_json(jsonstr)
+
+    # making the JSON File
+    def make_json(self, jsonstr):
+        try:
+            with open("phone_details.json","r") as myfile:
+                content = myfile.read()
+                if len(content) > 0:
+                    flag = 1
+                else:
+                    flag = 0
+        except:
+            content = ''
+            flag = 0
+
+        # checking if the phone already exists
+        if f'"{self.phone_name}"' not in content:
+            with open("phone_details.json","w+") as myfile:
+                if flag == 1:
+                    content = content[:-1]
+                    content = content + '\n,\n'+f'"{self.phone_name}" : ' + jsonstr + '\n}'
+                else:
+                    content = '{\n'+ f'"{self.phone_name}" : ' + jsonstr +'\n}'
+                myfile.write(content)
+        else:
+            print("You already have the phone details")
+        #Opening JSON file
+        self.open_file()
+            
+
+    def open_file(self):
```

Comment:
No class properties or state is being used. How about changing the method to a function or just using a decorator `@staticmethod` ?